src/testsuite/vec_f128_dummy.c

```diff
@@ -40,24 +40,177 @@
 //#define __DEBUG_PRINT__
 #include <pveclib/vec_f128_ppc.h>
 
+vui32_t
+test_and_bin128_2_vui32t (__binary128 f128, vui32_t mask)
+{
+  return vec_and_bin128_2_vui32t (f128, mask);
+}
+
+vui32_t
+test_andc_bin128_2_vui32t (__binary128 f128, vui32_t mask)
+{
+  return vec_andc_bin128_2_vui32t (f128, mask);
+}
+
+vui32_t
+test_xfer_bin128_2_vui32t (__binary128 f128)
+{
+  return vec_xfer_bin128_2_vui32t (f128);
+}
+
+vui128_t
+test_xfer_bin128_2_vui128t (__binary128 f128)
+{
+  return vec_xfer_bin128_2_vui128t (f128);
+}
+
+__binary128
+test_xfer_vui32t_2_bin128 (vui32_t f128)
+{
+  return vec_xfer_vui32t_2_bin128 (f128);
+}
+
 #ifndef PVECLIB_DISABLE_F128MATH
 #ifdef __FLOAT128__
+// TBD will sub-in pveclib softfloat for P8 when available
+
 __float128
 test_scalar_add128 (__float128 vra, __float128 vrb)
 {
-  return (vra + vrb);
+  if (__builtin_cpu_supports("ieee128"))
+    {
+      __float128 result;
+      __asm__(
+	      "xsaddqp %0,%1,%2"
+	      : "=v" (result)
+	      : "v" (vra), "v" (vrb)
+	      : );
+      return result;
+    }
+  else
+    // Generate call to __addkf3
+    return (vra + vrb);
```

Comment:
Showing my ignorance here, but why is this change needed? Will the compiler not do the right thing? (If not, why not?)

Reply:
Timing test to compare softfloat to hardfloat. Eventually vector softfloat for P8. This is half baked and will evolve. But is not production code and will not address now.